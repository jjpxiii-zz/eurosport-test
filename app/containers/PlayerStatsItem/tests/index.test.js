/**
 * Test the repo list item
 */

import React from 'react';
import { getByText, render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';

import { PlayerStatsItem } from '../index';

const renderComponent = (props = {}) =>
  render(
    <IntlProvider locale="en">
      <PlayerStatsItem {...props} />
    </IntlProvider>,
  );

describe('<PlayerStatsItem />', () => {
  let item;

  // Before each test reset the item data for safety
  beforeEach(() => {
    item = {
      firstname: 'Stan',
      lastname: 'Wawrinka',
      shortname: 'S.WAW',
      sex: 'M',
      country: {
        picture:
          'https://i.eurosport.com/_iss_/geo/country/flag/large/2213.png',
        code: 'SUI',
      },
      picture:
        'https://i.eurosport.com/_iss_/person/pp_clubteam/large/325225.jpg',
      data: {
        rank: 21,
        points: 1784,
        weight: 81000,
        height: 183,
        age: 33,
        last: [1, 1, 1, 0, 1],
      },
    };
  });

  it('should render a PlayerStatsItem', () => {
    const { container } = renderComponent({ item });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the player name', () => {
    const { container } = renderComponent({ item });
    expect(
      getByText(container, content => content.endsWith(item.lastname)),
    ).not.toBeNull();
  });

  it('should render the rank', () => {
    const { container } = renderComponent({ item });
    expect(
      getByText(container, content => content.endsWith(item.data.rank)),
    ).not.toBeNull();
  });

  it('should render the points', () => {
    const { container } = renderComponent({ item });
    expect(
      getByText(container, content => content.endsWith(item.data.points)),
    ).not.toBeNull();
  });

  it('should render the age', () => {
    const { container } = renderComponent({ item });
    expect(
      getByText(container, content => content.endsWith(item.data.age)),
    ).not.toBeNull();
  });

  it('should render the wins and losses', () => {
    const { container } = renderComponent({ item });
    expect(
      getByText(container, content => content.endsWith('4W 1L')),
    ).not.toBeNull();
  });

  it('should render the height', () => {
    const { container } = renderComponent({ item });
    expect(
      getByText(container, content => content.endsWith(item.data.height)),
    ).not.toBeNull();
  });

  it('should render the weight', () => {
    const { container } = renderComponent({ item });
    expect(
      getByText(container, content =>
        content.endsWith(Math.round(item.data.weight / 1000, 2)),
      ),
    ).not.toBeNull();
  });

  it('should render the flag', () => {
    const { container } = renderComponent({ item });
    expect(container.querySelector('img')).not.toBeNull();
  });
});
