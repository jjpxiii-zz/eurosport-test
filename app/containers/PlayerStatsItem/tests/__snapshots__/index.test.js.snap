// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerStatsItem /> should render a PlayerStatsItem 1`] = `
<tr
  class="Wrapper-p1o7x3-0 bHJuyz"
>
  <td>
    <img
      alt="Stan Wawrinka"
      src="https://i.eurosport.com/_iss_/person/pp_clubteam/large/325225.jpg"
    />
  </td>
  <td>
    <img
      alt="SUI"
      class="CountryIcon-xrr94z-0 exvXay"
      src="https://i.eurosport.com/_iss_/geo/country/flag/large/2213.png"
    />
  </td>
  <td>
    Stan
     
    Wawrinka
  </td>
  <td>
    21
  </td>
  <td>
    1784
  </td>
  <td>
    183
  </td>
  <td>
    <span>
      81
    </span>
  </td>
  <td>
    33
  </td>
  <td>
    4
    W 
    1
    L
  </td>
</tr>
`;
